### DISCUSSION TITLE
Doubt regarding the liquidate function in lesson 17 of Stablecoin

### DISCUSSION THREAD

**Discussion by heheboii11** [Cyfrin/foundry-full-course-cu#2845] [Q&A]
Hey Everyone, while going through lesson 17 of stablecoin. I noticed that when a one User wants to liquidate another user if their ```healthfactor < 1``` , 

Suppose,
If the user's collateral is of 3 tokens( Eg: WETH, WBTC, WSOL) and they have deposited each token worth ```$100```, so this makes ```$300``` in collateral and they mints 150 dollars of ```DSC```, so when the other user calls the liquidate function, in the code it only transfers one token and does not do anything about the other tokens, leaving the user who tried to liquidate in loss. This logic what I am thinking is in the correct way or is it wrong ?

```
And Patrics also mentioned there is a bug in the code, is it the above mentioned one or there is other bug?.
```

**Comment by EngrPips** [comment]
Hello @heheboii11, What I can say is that the codebase lacks many necessary parts that a normal protocol codebase will have that make it so it is only to teach what a stablecoin is. There can be more bugs in the codebase as it is just a tutorial codebase.

**Reply by heheboii11** [reply]
Ahh, sounds good!!. Is this the bug patrics is talking about ?

**Reply by EngrPips** [reply]
Actually `No` when you run into that bug you will open a discussion for it trust me because it will make your test fail.

**Reply by heheboii11** [reply]
looking forward to it and I am actually refactoring liquidate function to the idea I said above and will see how much time it will take !!

**Reply by EngrPips** [reply]
Alright, Friend. looking forward to hearing from you.
